[abseil] Add -fPIC option

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **abseil/20180600@bincrafters/stable**
  * Operating System+version: **Debian 9 (stretch) amd64**
  * Compiler+version: **gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516**
  * Conan version: **Conan 1.18.5**
  * Python version: **Python 3.5.3**

### Description of your feature

While trying to package Google's or-tools, I've hit on error due to lack of `-fPIC`:
```
[ 79%] Linking CXX shared library libortools.so                                                                                      
/usr/bin/ld: /home/vincas/.conan/data/abseil/20180600/bincrafters/stable/package/dbc1e3cfd311e524aae72619c1b8bf14a12a97b5/lib/libabsl_strings.a(charconv_bigint.cc.o): r
elocation R_X86_64_PC32 against symbol `_ZN4absl16strings_internal10kFiveToNthE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value                                                              
collect2: error: ld returned 1 exit status                                                                                                                              
CMakeFiles/ortools.dir/build.make:539: recipe for target 'libortools.so.7.3.9999' failed 
```

So it seems it would be useful to have this option, also maybe shared build too?

--- a/conanfile.py
+++ b/conanfile.py
@@ -22,8 +22,8 @@ class AbseilConan(ConanFile):
     generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
     requires = "cctz/2.2@bincrafters/stable"
-    options = {"cxx_standard": [11, 14, 17], "build_testing": [True, False]}
-    default_options = {"cxx_standard": 11, "build_testing": False}
+    options = {"cxx_standard": [11, 14, 17], "build_testing": [True, False], "fPIC" : [True, False]}
+    default_options = {"cxx_standard": 11, "build_testing": False, "fPIC": True}
     short_paths = True
     _source_subfolder = "source_subfolder"
 
@@ -32,6 +32,10 @@ class AbseilConan(ConanFile):
         extracted_dir = "abseil-cpp-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
         tools.patch(patch_file="leak_check.patch", base_path=self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
         if self.settings.os == "Windows" and \
@@ -53,6 +57,8 @@ class AbseilConan(ConanFile):
         self.copy("*.inc", dst="include", src=self._source_subfolder)
         self.copy("*.a", dst="lib", src=".", keep_path=False)
         self.copy("*.lib", dst="lib", src=".", keep_path=False)
+        self.copy("*.so", dst="lib", src=".", keep_path=False)
+        self.copy("*.dll", dst="bin", src=".", keep_path=False)
 
     def package_info(self):
         if self.settings.os == "Linux":

